homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,137 @@
+        self.char_target_heat_cool.value = 0
+        self.char_target_heat_cool.setter_callback = self.set_heat_cool
+
+        self.char_current_temp = self.service_thermostat. \
+            get_characteristic(CHAR_CURRENT_TEMPERATURE)
+        self.char_current_temp.value = 0.0
+        self.char_target_temp = self.service_thermostat. \
+            get_characteristic(CHAR_TARGET_TEMPERATURE)
+        self.char_target_temp.value = 0.0
+        self.char_target_temp.setter_callback = self.set_temperature_debounced
+
+        self.char_display_units = self.service_thermostat. \
+            get_characteristic(CHAR_TEMP_DISPLAY_UNITS)
+        self.char_display_units.value = 0
+
+    def run(self):
+        """Method called be object after driver is started."""
+        state = self._hass.states.get(self._entity_id)
+        self.update_thermostat(new_state=state)
+
+        async_track_state_change(self._hass, self._entity_id,
+                                 self.update_thermostat)
+
+    def set_heat_cool(self, value):
+        """Move operation mode to value if call came from HomeKit."""
+        if value in HC_HOMEKIT_TO_HASS:
+            _LOGGER.debug("%s: Set heat-cool to %d", self._entity_id, value)
+            self.heat_cool_flag_target_state = True
+            hass_value = HC_HOMEKIT_TO_HASS[value]
+            self._hass.services.call('climate', 'set_operation_mode',
+                                     {ATTR_ENTITY_ID: self._entity_id,
+                                      ATTR_OPERATION_MODE: hass_value})
+
+    def set_temperature_debounced(self, value):
+        """Set target temperature to value if call came from HomeKit."""
+        _LOGGER.debug("%s: Call to set temperature to %.2f",
+                      self._entity_id, value)
+
+        def perform_call():
+            """Perform the set_target_temperature function call."""
+            self.set_target_temperature(value)
+        # Cancel timer if one was running
+        if self._call_timer is not None:
+            try:
+                self._call_timer.cancel()
+            except AttributeError:
+                pass
+        self._call_timer = Timer(1, perform_call)
+        self._call_timer.start()
+
+    def set_target_temperature(self, value):
+        """Set target temperature to value if call came from HomeKit."""
+        _LOGGER.debug("%s: Set target temperature to %.2f",
+                      self._entity_id, value)
+        self.temperature_flag_target_state = True
+        self._hass.services.call(
+            'climate', 'set_temperature',
+            {ATTR_ENTITY_ID: self._entity_id,
+             ATTR_TEMPERATURE: value})
+
+    def update_thermostat(self, entity_id=None,
+                          old_state=None, new_state=None):
+        """Update security state after state changed."""
+        if new_state is None:
+            return
+        current_temp = new_state.attributes[ATTR_CURRENT_TEMPERATURE]
```

Comment:
Add an empty line to improve readability. Also between currently L118/L119 and L127/L128.